src/Meziantou.Analyzer/Rules/UseShellExecuteAnalyzer.cs

```diff
@@ -0,0 +1,112 @@
+
+            ctx.RegisterOperationAction(analyzerContext.AnalyzeInvocation, OperationKind.Invocation);
+            ctx.RegisterOperationAction(analyzerContext.AnalyzeObjectCreation, OperationKind.ObjectCreation);
+        });
+    }
+
+    private sealed class AnalyzerContext(Compilation compilation)
+    {
+        private readonly INamedTypeSymbol? _processStartInfoSymbol =
+            compilation.GetBestTypeByMetadataName("System.Diagnostics.ProcessStartInfo")!;
+
+        private readonly INamedTypeSymbol? _processSymbol =
+            compilation.GetBestTypeByMetadataName("System.Diagnostics.Process")!;
+
+        public bool IsValid => _processStartInfoSymbol is not null;
+
+        public void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            var operation = (IInvocationOperation)context.Operation;
+            if (IsProcessStartInvocation(operation))
+            {
+                if (!operation.Arguments.Any(IsProcessStartInfo))
+                {
+                    // Calling Process.Start without ProcessStartInfo
+                    context.ReportDiagnostic(UseProcessStartOverload, operation.Syntax);
+                }
+            }
+        }
+
+        public void AnalyzeObjectCreation(OperationAnalysisContext context)
+        {
+            var operation = (IObjectCreationOperation)context.Operation;
+            if (IsProcessStartInfoCreation(operation))
+            {
+                if (operation.Syntax is ObjectCreationExpressionSyntax { Initializer: {} initializer } )
+                {
+                    var useShellExecuteInitializer = initializer.Expressions.OfType<AssignmentExpressionSyntax>()
+                        .FirstOrDefault(x => x.Left is IdentifierNameSyntax
+                        {
+                            Identifier.Text: "UseShellExecute"
+                        });
+
+                    if (useShellExecuteInitializer is null)
+                    {
+                        // Constructing ProcessStartInfo without setting UseShellExecute in the initializer
+                        context.ReportDiagnostic(UseShellExecuteMustBeExplicitlySet, operation.Syntax);
+                    }
+                }
+                else
+                {
+                    // Constructing ProcessStartInfo with not initializer at all
+                    context.ReportDiagnostic(UseShellExecuteMustBeExplicitlySet, operation.Syntax);
+                }
+            }
+        }
+
+        private bool IsProcessStartInfo(IArgumentOperation operation)
+            => operation.Value.Type.IsEqualTo(_processStartInfoSymbol);
+
+        private bool IsProcessStartInfoCreation(IObjectCreationOperation operation)
+            => operation.Type.IsEqualTo(_processStartInfoSymbol);
+
+        private bool IsProcessStartInvocation(IInvocationOperation operation)
+            => operation.TargetMethod.Name == "Start" && operation.TargetMethod.ContainingType.IsEqualTo(_processSymbol);
```

Comment:
I think this also detects the instance method `process.Start()`. You should validate the method is `static`.